main.rb

```diff
@@ -0,0 +1,152 @@
+module Enumerable
+  def my_each
+    return to_enum unless block_given?
+
+    array = to_a
+    index = 0
+    while index < size
+      yield(array[index])
+      index += 1
+    end
+    array
+  end
+
+  #--------my each by index -------
+  def my_each_with_index
+    return to_enum unless block_given?
+
+    array = to_a
+    index = 0
+    while index < size
+      yield(array[index], index)
+      index += 1
+    end
+    array
+  end
```

Comment:
It should have a working my each with index method. This method should work exactly like each with index 
![my_each_with_index](https://user-images.githubusercontent.com/32361963/95429711-6e284300-0953-11eb-8d9b-f289ad2d8843.png)

The problem in this method is that when a block is given when self is a range it should return the range itself after calling the given block once for each element in self.